mobx_codegen/lib/src/mobx_codegen_base.dart

```diff
@@ -20,18 +20,28 @@ import 'package:mobx_codegen/src/template/util.dart';
 import 'package:source_gen/source_gen.dart';
 
 class StoreGenerator extends Generator {
-  final _storeChecker = const TypeChecker.fromRuntime(Store);
+  final _storeMixinChecker = const TypeChecker.fromRuntime(Store);
+  final _storeAnnotationChecker = const TypeChecker.fromRuntime(MakeStore);
 
   @override
   FutureOr<String> generate(LibraryReader library, BuildStep buildStep) {
-    return library.classes
-        .where(isValidStoreClass)
-        .expand((c) => expandToMixinCode(library, c))
-        .toSet()
-        .join('\n\n');
+    final generatedMixins = library.classes
+        .where(isValidMixinStoreClass)
+        .expand((c) => generateCodeForMixinStore(library, c));
+    final generatedSubclasses = library.classes
```

Comment:
Will need a check to ensure only one of `@store` of `Store` mixin is used. We should throw an error and fail the codegen if both are detected on the same class. The pattern we are following is in [errors.dart](https://github.com/mobxjs/mobx.dart/blob/master/mobx_codegen/lib/src/errors.dart)

Reply:
Right, sorry. I got excited. Give me a few.

Reply:
Done